torch/fx/passes/infra/op_expander_pass.py

```diff
@@ -0,0 +1,63 @@
+from typing import Callable, Generic, Optional, TypeVar
+
+import torch
+
+
+T = TypeVar("T")
+
+__all__ = ["OpExpanderPass"]
+
+
+class OpExpanderPass(Generic[T]):
+    def __init__(
+        self,
+        pattern_matcher: Callable[[torch.fx.Node], tuple[bool, Optional[T]]],
+        node_expander: Callable[
+            [torch.fx.Node, Optional[T], torch.fx.GraphModule], torch.fx.Node
+        ],
+        extra_filter: Optional[Callable[[torch.fx.Node], bool]] = None,
+    ):
+        """
+        Initialize the OpExpander pass.
+
+        Args:
+            pattern_matcher: Returns (matched: bool, data: Optional[T])
+            node_expander: Function to expand a node using optional match data of type T
+            extra_filter: Optional extra filtering criteria for matching nodes
```

Comment:
@eellison do we have anything like this already? If not I'm happy to have something like this in PyTorch

Reply:
Patrick tells me the semantics are:
- for each node, check if pattern_matcher(node) returns True
- if it returns True, then execute node_expander to produce a graph
- replace the node with the produced graph

Seems kind of like our pattern matcher but not really.

Reply:
Yea, seems a bit duplicative with pattern matcher..